### PR TITLE
Adding loss classes for A2C

### PR DESCRIPTION
**Pull request by NimishaDey**
## Description

Adding A2C Discrete and A2C Continuous loss classes for A2C.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [ x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/rl/torch_rl/torch_a2c.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import numpy as np
+from typing import List
+try:
+    import torch
+    has_pytorch = True
+except:
+    has_pytorch = False
+
+
+class A2CLossDiscrete(object):
```
Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:11]: Can you add this to the docs?

Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:11]: Do the losses inherit from object in general? Check the other deepchem losses

File: deepchem/rl/torch_rl/torch_a2c.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    """This class computes the loss function for A2C with discrete action spaces.
+    The A2C algorithm optimizes all outputs at once using a loss that is the sum of three terms:
+    1. The policy loss, which seeks to maximize the discounted reward for each action.
+    2. The value loss, which tries to make the value estimate match the actual discounted reward
+        that was attained at each step.
+    3. An entropy term to encourage exploration.
```
Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:17]: Spacing

File: deepchem/rl/torch_rl/torch_a2c.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        policy_loss: torch.Tensor = -torch.mean(
+            advantage * torch.sum(action * log_prob, 1))
+        value_loss: torch.Tensor = torch.mean(
+            torch.square(torch.from_numpy(reward) - value))
+        entropy: torch.Tensor = -torch.mean(torch.sum(prob * log_prob, 1))
+        return policy_loss + self.value_weight * value_loss - self.entropy_weight * entropy
+
+
+# Note: For continuous case, when an additional loss term correspoding to mean is calculated only then the gradients get calculated for the layers of the model.
+class A2CLossContinuous(object):
```
Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:72]: Add to docs

File: deepchem/rl/torch_rl/torch_a2c.py [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    """This class computes the loss function for A2C with continuous action spaces.
+    Example
```
Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:74]: Spacing

File: deepchem/rl/torch_rl/torch_a2c.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
+"""Advantage Actor-Critic (A2C) algorithm for reinforcement learning."""
+import numpy as np
+from typing import List
+try:
+    import torch
+    has_pytorch = True
+except:
+    has_pytorch = False
+
```
Comment (rbharath) [inline deepchem/rl/torch_rl/torch_a2c.py:9]: We need to add unit tests for these new classes.

**Review by rbharath — COMMENTED** [review]
To summarize, we need tests/docs

File: deepchem/rl/torch_rl/torch_a2c.py (hunk 1 above)
Comment (NimishaDey) [reply deepchem/rl/torch_rl/torch_a2c.py:11]: Added

File: deepchem/rl/torch_rl/torch_a2c.py (hunk 3 above)
Comment (NimishaDey) [reply deepchem/rl/torch_rl/torch_a2c.py:76]: Added

File: deepchem/rl/torch_rl/torch_a2c.py (hunk 1 above)
Comment (NimishaDey) [reply deepchem/rl/torch_rl/torch_a2c.py:11]: The tensorflow implementation of these losses does inherit from object.

**Review by rbharath — APPROVED** [review]
LGTM
